src/utils/geometry/filterPoints.ts

```diff
@@ -0,0 +1,61 @@
+import { Image } from '../..';
+
+import { Point } from './points';
+
+interface FilterPointsOptions {
+  /**
+   * The number of points that should be removed if they are close to extremum.
+   */
+  removeClosePoints: number;
+  /**Shows what kind of extremum is being computed */
+  kind: 'minimum' | 'maximum';
+}
+/**
+ * Finds extreme values of an image which are not stacked together.
+ *
+ * @param points - Array of points that should be combined to improve.
+ * @param image - Image which extrema are calculated from.
+ * @param options - FilterPointsOptions
+ * @returns Array of Points.
+ */
+export function filterPoints(
+  points: Point[],
+  image: Image,
+  options: FilterPointsOptions,
+) {
+  let { removeClosePoints = 0, kind = 'maximum' } = options;
+  const isMax = kind === 'maximum';
+
+  let sortedPoints = points
+    .sort((a, b) => {
+      return Math.min(
+        image.getValue(a.column, a.row, 0),
+        image.getValue(b.column, b.row, 0),
+      );
+    })
+    .reverse();
```

Comment:
Just make a conditional reverse depending if it is min or max. This will avoid to have to add a condition in the loop.